src/main/java/com/example/demo/services/UserService.java

```diff
@@ -0,0 +1,82 @@
+package com.example.demo.services;
+
+import com.example.demo.dtos.ProductDTO;
+import com.example.demo.entities.User;
+import com.example.demo.mappers.DtoMapper;
+import com.example.demo.products.Product;
+import com.example.demo.repositories.UserRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Stream;
+
+@Service
+@RequiredArgsConstructor
+public class UserService {
+
+    @Autowired
+    public static UserRepository userRepository;
+    private final DtoMapper mapper;
```

Comment:
Mapper should belong to the controller. The best way is to keep separate mappers for every DTO object (e.g. ProductMapper), each of them to hold 2 methods: from DTO to Model, from Model to DTO.

Reply:
Okay, mapper now belongs to the controller only but what about separating them there is a problem since there are different classes extending Product class and if I would separate mappers I would get many mapper dependencies in controller BUT they are being used only in one context (in one method) so I created one mapper for all. Should I separate them?